mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageZipper.java

```diff
@@ -32,21 +34,33 @@
 import org.apache.james.mailbox.backup.MessageIdExtraField;
 import org.apache.james.mailbox.backup.SizeExtraField;
 import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.util.OptionalUtils;
+import org.reactivestreams.Publisher;
 
 import com.google.common.annotations.VisibleForTesting;
 
-class DeletedMessageZipper {
+import reactor.core.publisher.Mono;
+
+public class DeletedMessageZipper {
+
+    interface DeletedMessageContentLoader {
+        Publisher<InputStream> load(DeletedMessage deletedMessage);
+    }
 
     DeletedMessageZipper() {
         ExtraFieldUtils.register(MessageIdExtraField.class);
         ExtraFieldUtils.register(SizeExtraField.class);
     }
 
-    void zip(Stream<DeletedMessageWithContent> deletedMessages, OutputStream outputStream) throws IOException {
+    public void zip(DeletedMessageContentLoader contentLoader, Stream<DeletedMessage> deletedMessages,
+                    OutputStream outputStream) throws IOException {
         try (ZipArchiveOutputStream zipOutputStream = newZipArchiveOutputStream(outputStream)) {
 
             AtomicReference<IOException> failureEncountered = new AtomicReference<>();
```

Comment:
Why do we need the `AtomicReference` for?

Can't we just sneakyThrow?

Reply:
precisely!